src/test/java/org/folio/integration/http/VertxOkapiHttpClientTest.java

```diff
@@ -54,6 +55,30 @@ void afterEach() {
     stopVertx();
   }
 
+  @SneakyThrows
+  @Test
+  void throwsTimeoutWhenResponseTakesOverFiveSeconds() {
+    final var getEndpoint = matchingFolioHeaders(get(urlPathEqualTo("/record")));
+
+    fakeWebServer.stubFor(getEndpoint.willReturn(ok()
+      .withFixedDelay(6000)
+      .withBody(dummyJsonResponseBody())
+      .withHeader("Content-Type", "application/json")));
+
+    final var client = createClient();
+
+    final var getCompleted = client.get(
+      "/record", Headers.toMap(fakeWebServer.baseUrl()));
+
+    Exception exception = assertThrows(ExecutionException.class, ()->{
+      getCompleted.get(6, SECONDS);
+    });
+
+    String err = "The timeout period of 5000ms has been exceeded";
+    assertThat(exception.getMessage().indexOf(err), not(-1));
```

Comment:
Is this only part of the error message?

If so, is there a hamcrest matcher that checks a string contains a sub-string? 

If there is, I think that might be more readable than this.

Reply:
It is.  The actual error message contains a lot of data and is extremely long.  I'll check and see if I can find a hamcrest matcher.